### DISCUSSION TITLE
Syntax Support FalconRTR

### DISCUSSION THREAD

**Discussion by Cgiancasproideos** [CrowdStrike/psfalcon#334] [Q&A]
HI,
i checked the documentation, but i couldn't fix my error, below i have two examples that always give me error.
EXAMPLE 1: 
Invoke-FalconRTR -command runscript -Argument -Raw=```reg delete "HKLM\Software\Wow6432Node\Pulse Secure\Device Id" /v "DeviceId" /f``` -GroupID 6bd9d6ae8409452a9a9be63e4b4a3f58
EXAMPLE 2 
$commandarguments = gwmi Win32_Product -filter "name like 'Cisco%'" | % { $_.Uninstall() }
Invoke-FalconRTR -command runscript -Argument $commandarguments -hostid 283b4f3ae28f4f50a891dbf244dead5e -Timeout 30 -QueueOffline $False
the third example does not give me an error and the restart of the machines works regularly.
EXAMPLE 3 - OK:
Invoke-FalconRtr -Command Restart -Argument -Confirm -GroupId 4046b127fd52421e989137061084ec9f

**Comment by bk-cs** [comment]
Example 1:
```powershell
Invoke-FalconRtr -command runscript -Argument '-Raw=```reg delete "HKLM\Software\Wow6432Node\Pulse Secure\Device Id" /v "DeviceId" /f```' -GroupId 6bd9d6ae8409452a9a9be63e4b4a3f58
```
You were likely having trouble because of a lack of quotes around the `Argument` string.

Example 2:
```powershell
Invoke-FalconRtr -command runscript -Argument '-Raw=```gwmi Win32_Product -filter "name like `"Cisco%`"" | % { $_.Uninstall() }```' -HostId <id> -Timeout 30 -QueueOffline $False
```
This one is trickier because of two sets of double quotes -- one of them needs to be escaped.

The easiest way to remember formatting:

1. `Command` is the command you're running
2. `Argument` is everything that would come over the command _enclosed in a single string_
3. Anything with a space in it needs to be quoted twice

This Microsoft article can help explain how quoting works in PowerShell: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.3

**Reply by Cgiancasproideos** [reply]
Hi,
thanks for your support. I tryed the string successfully.
Best regards
